resources/images/s3/filestorage.go

```diff
@@ -112,9 +117,29 @@ func NewSimpleStorageServiceDefaults(bucket, region string) (*SimpleStorageServi
 	}, nil
 }
 
+func getImageByTenant(ctx context.Context, objectID string, tenantExpected bool) (string, error) {
```

Comment:
how about making this a method on `(s *SimpleStorageService)`? then you get rid of the awkward `tenantExpected` thing, you already have this info in `s`.

ignore if there's a reason for keeping this separate.